app/caldav/index.ts

```diff
@@ -259,11 +267,51 @@ export async function getTakenTimeSlots(
       <?xml version="1.0"?>
       <c:free-busy-query xmlns:c="urn:ietf:params:xml:ns:caldav">
         <c:time-range
-          start="${formatDateString(start)}"
-          end="${formatDateString(end)}"
+          start="${formatDateAsICS(start)}"
+          end="${formatDateAsICS(end)}"
         />
       </c:free-busy-query>`.trim(),
   })
 
   return await icsFreeBusyToInternalFreeBusy(response.data.toString())
 }
+
+interface EventDetails {
+  name: string
+  timezone: number
+  start: Date
+  end: Date
+  location: string
+  description: string
+}
+
+export async function createEvent(calendar: CalendarConnectionDetails, eventDetails: EventDetails) {
+  const dateNow = new Date()
+  const uid = uuidv4()
+  const data = `BEGIN:VCALENDAR
+VERSION:2.0
+PRODID:-//MailClient.VObject/8.0.3385.0
+BEGIN:VEVENT
+UID:${uid}
+DTSTART;TZID=Europe/Berlin:${formatDateAsICS(eventDetails.start)}
+DTEND;TZID=Europe/Berlin:${formatDateAsICS(eventDetails.end)}
+TRANSP:OPAQUE
+X-MICROSOFT-CDO-BUSYSTATUS:BUSY
+LAST-MODIFIED:${formatDateAsICS(dateNow)}
+DTSTAMP:${formatDateAsICS(dateNow)}
+CREATED:${formatDateAsICS(dateNow)}
+LOCATION:${eventDetails.location}
+SUMMARY:${eventDetails.name}
+CLASS:PUBLIC
+END:VEVENT
+END:VCALENDAR\r\n`
+  const response = await makeRequestTo(calendar, {
+    data: data,
+    method: "PUT",
+    urlExtension: uid + ".ics",
+    headers: {
+      Depth: 1,
+    },
+  })
+  return response.res.statusMessage
```

Comment:
```suggestion
  return response.res.statusMessage === "Created" ? "success" : "failure"
```
The exact value of `statusMessage` is an implementation detail local to this function and could potentially differ between CalDav implementations.
Thus, the evaluation of wether a request was successful or not should be performed by _this_ function, and not by the caller.

Reply:
makes sense :+1: